cli/src/config.rs

```diff
@@ -335,13 +350,15 @@ impl ConfigEnv {
     #[instrument]
     pub fn reload_user_config(&self, config: &mut RawConfig) -> Result<(), ConfigLoadError> {
         config.as_mut().remove_layers(ConfigSource::User);
-        if let Some(path) = self.existing_user_config_path() {
-            if path.is_dir() {
-                config.as_mut().load_dir(ConfigSource::User, path)?;
-            } else {
-                config.as_mut().load_file(ConfigSource::User, path)?;
-            }
-        }
+        self.existing_user_config_paths()
+            .iter()
+            .try_for_each(|path| {
+                if path.is_dir() {
+                    config.as_mut().load_dir(ConfigSource::User, path)
+                } else {
+                    config.as_mut().load_file(ConfigSource::User, path)
+                }
+            })?;
```

Comment:
nit: `?; Ok(())` can be omitted. It's also good to use plain `for` loop.

Reply:
I've switched to the `for` loop, but removing the `Ok(())` would require removing the `Result` return type which would cascade elsewhere---is that okay?

Reply:
No, don't remove `Result`. I meant either `for` loop or `.try_for_each()` without `Ok(())`.